baybe/targets/binary.py

```diff
@@ -0,0 +1,91 @@
+"""Binary targets."""
+
+from typing import TypeAlias
+
+import numpy as np
+import pandas as pd
+from attrs import define, field
+from attrs.validators import instance_of
+
+from baybe.exceptions import InvalidTargetValueError
+from baybe.serialization import SerialMixin
+from baybe.targets.base import Target
+
+ChoiceValue: TypeAlias = bool | int | float | str
+"""Types of values that a :class:`BinaryTarget` can take."""
+
+_POSITIVE_VALUE_COMP = True
+"""Computational representation of the positive choice value."""
+
+_NEGATIVE_VALUE_COMP = False
+"""Computational representation of the negative choice value."""
+
+
+@define(frozen=True)
+class BinaryTarget(Target, SerialMixin):
+    """Class for binary targets."""
+
+    # FIXME[typing]: https://github.com/python-attrs/attrs/issues/1336
+
+    positive_value: ChoiceValue = field(
```

Comment:
@AVHopp, @Scienfitz: I was not sure what is the best name for these attributes. Alternatives could be `success_value`/`failure_value` or similar. Potentially even just `heads`/`tails` but then we lose the direction. Shall we keep it / change it? Input welcome

Reply:
`positive` implies theres also a negative, when thinking about bernulli experiments the value for failure however is conventionally 0 and not negative

success and failure already imply an interpretation of the result which might not be true

why not give this target a field `values` validated to have exactly two members?

Reply:
Hm, have also thought about that but also has a couple of downsides:
* You cannot change value without also providing the other
* Most important: It is not clear from the interface what is the optimization direction. I.e. is the first or the second the "good" value? If that problem didn't exist, I could also go for heads/tails
* If you group in to a single `values` attribute, it would not be a bit inconsistent with other places. E.g. we also don't group the alpha and beta of the beta prior, for the same reasons above

Reply:
So ideally, we could come up with a terminology that just avoids the "negative" association...

Reply:
yeah after trying to come up with something more neutral and failing I agree success / failure is prob the best even though it also has interpretation but thats pretty much always what a binary target will very often be

Reply:
done

Reply:
changed to `success`/`failure`